lib/taskana-core/src/main/java/pro/taskana/impl/TaskServiceImpl.java

```diff
@@ -407,27 +431,46 @@ static void setPrimaryObjRef(TaskImpl task) {
         task.setPrimaryObjRef(objRef);
     }
 
-    private void validatePrimaryObjectReference(TaskImpl task) throws InvalidArgumentException {
+    private void validatePrimaryObjectReference(ObjectReference primObjRef, String objName)
```

Comment:
This is a generic validation? If it is, maybe the "primary" should be removed from the name?